src/ast/passes/codegen_llvm.cpp

```diff
@@ -769,16 +774,17 @@ void CodegenLLVM::visit(Call &call)
       auto &arg = *call.vargs->at(0);
       fixed_buffer_length = arg.type.GetNumElements() *
                             arg.type.GetElementTy()->GetSize();
-      length = b_.getInt8(fixed_buffer_length);
+      length = b_.getInt32(fixed_buffer_length);
     }
 
+    Value *scratch_buf = b_.CreateGetStrScratchMap(str_id_, nullptr, call.loc);
```

Comment:
nit: why not just do the increment in place in a post op (seems less error prone)

Reply:
Yeah helper is probably hardest to get wrong later. I'll tack on a commit to do this

Reply:
Looking closer, all the other IDs are incremented like this. Probably better to keep consistency. Maybe we can create an X-macro to generate helpers as well as save/restore in create_reset_ids()?

Reply:
> Looking closer, all the other IDs are incremented like this.

I see what you mean but I can't seem to find any reason why. In some cases the increment happens on the next line.  Anyway, I can look into cleaning that up later. No need to block this PR

Reply:
Yeah, no reason why it has to be like this. Would be great if you could take a stab at improving this